### PR TITLE
SNOW-644959: Return all the extra columns in the insertRow response

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
- return all the extra columns in the insertRow response, this is needed by the KC schematization work since we want to add all the columns at once during schema evolution
- Clean up the HttpConnectionManagerDaemonThread during client close
- Passing the correct double quotes column name to EP

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java [hunk 1]
```diff
@@ -628,29 +628,41 @@ private String formatColumnName(String columnName) {
...
-      Field field = this.fields.get(columnName);
-      if (field == null) {
-        throw new SFException(
-            ErrorCode.INVALID_ROW,
-            "Extra column: " + columnName,
-            "Columns not present in the table shouldn't be specified.");
+    if (!extraCols.isEmpty()) {
+      throw new SFException(
+          ErrorCode.INVALID_ROW,
+          "Extra columns: " + extraCols,
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:646]: Arrays.toString(extraCols)?

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java [hunk 2]
```diff
@@ -628,29 +628,41 @@ private String formatColumnName(String columnName) {
...
+      if (!inputColNamesMap.containsKey(columnName)) {
+        missingCols.add(columnName);
       }
     }
 
-    return inputColumns;
+    if (!missingCols.isEmpty()) {
+      throw new SFException(
+          ErrorCode.INVALID_ROW,
+          "Missing columns: " + missingCols,
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:661]: Arrays.toString(list.toArray())

File: src/test/java/net/snowflake/ingest/streaming/internal/RowBufferTest.java [hunk 3]
```diff
@@ -1485,6 +1485,27 @@ private void testMissingColumnCheckHelper(SnowflakeStreamingIngestChannelInterna
...
+    colBoolean.setScale(0);
+
+    innerBuffer.setupSchema(Collections.singletonList(colBoolean));
+    Map<String, Object> row = new HashMap<>();
+    row.put("COLBOOLEAN1", true);
+    row.put("COLBOOLEAN2", true);
+    row.put("COLBOOLEAN3", true);
+
+    InsertValidationResponse response = innerBuffer.insertRows(Collections.singletonList(row), "1");
+    Assert.assertTrue(response.hasErrors());
```
Comment (sfc-gh-japatel) [inline src/test/java/net/snowflake/ingest/streaming/internal/RowBufferTest.java:1506]: can you assert the error too?

File: src/test/java/net/snowflake/ingest/streaming/internal/RowBufferTest.java [hunk 4]
```diff
@@ -1485,6 +1485,27 @@ private void testMissingColumnCheckHelper(SnowflakeStreamingIngestChannelInterna
     }
   }
 
+  @Test
+  public void testExtraColumnCheck() {
```
Comment (sfc-gh-japatel) [inline src/test/java/net/snowflake/ingest/streaming/internal/RowBufferTest.java:1489]: nit: `testExtraColumnsCheck`

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java [hunk 5]
```diff
@@ -628,29 +628,41 @@ private String formatColumnName(String columnName) {
...
-        throw new SFException(
-            ErrorCode.INVALID_ROW,
-            "Missing column: " + columnName,
-            "Values for all non-nullable columns must be specified.");
+    // Check for extra columns in the row
+    List<String> extraCols = new ArrayList<>();
+    for (String columnName : inputColNamesMap.keySet()) {
+      Field field = this.fields.get(columnName);
+      if (field == null) {
+        extraCols.add(inputColNamesMap.get(columnName));
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:639]:
```suggestion
      if (!this.fields.containsKey(columnName)) {
        extraCols.add(inputColNamesMap.get(columnName));
```

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm! Minor suggestions

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java (hunk 1 above)
Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:645]: I actually used the similar way (list.toString()), removing the toString is actually suggested by Intellij

**Pull request merged**